frontend/src/stores/ParticipantStore.js

```diff
@@ -1,49 +1,60 @@
-import { observable, action, flow, toJS } from "mobx"
-import participantApi from "../api/participantApi"
+import { observable, action, flow, toJS, decorate } from "mobx"
+import { createContext } from "react"
+import api from "../api"
 
-class ParticipantStore {
+export class ParticipantStore {
   constructor(rootStore) {
     this.rootStore = rootStore
   }
 
-  //participants = observable([])
-  @observable participants = []
+  // Store Params
+  participants = []
+  params = {}
 
-  @action setParticipant(participant, index) {
-    this.participants[index] = participant
+  // Setters
+  setParticipantsList = data => {
+    this.participants = data
+  }
+  setUserIdParam = data => {
+    this.params.pp_id = data
+  }
+  setFirstNameParam = data => {
+    this.params.first_name = data
+  }
+  setLastNameParam = data => {
+    this.params.last_name = data
+  }
+
+  // Getters
+  getParticipantsList = () => {
+    return toJS(this.participants)
+  }
+  getParams = () => {
+    return toJS(this.params)
   }
 
```

Comment:
Why are you converting these to JSON?

Reply:
This makes sense if we are storing the data as an observable in the store variables and only converting it to JS on a get call.

Reply:
It makes sense if the value in the store is an observable there by needing the JS conversion for the component

Reply:
Let's discuss it tonight